ios/brave-ios/Sources/Brave/Frontend/Browser/HomePanel/NTPDataSource.swift

```diff
@@ -195,6 +197,13 @@ public class NTPDataSource {
     Preferences.NewTabPage.backgroundRotationCounter.value += 1
 
     guard let bgWithIndex = backgroundSet[safe: backgroundIndex] else { return nil }
+
+    if bgWithIndex.backgroundVideoPath != nil,
+      Preferences.NewTabPage.backgroundMediaType != .sponsoredImagesAndVideos
+    {
+      return NTPWallpaper.image(.fallback)
+    }
```

Comment:
What is this extra logic for?

Reply:
If `bgWithIndex` is a video background but user didn't select `Sponsored Images & Videos` option in settings, then we fallback to default NTPWallpaper.

So it allows not to show video background when option is not selected.

Reply:
I guess Im confused why this edge case should be possible to hit. Shouldn't we be filtering the backgrounds to the appropriate list based on the pref?

Reply:
makes sense, I will try to do filtering

Reply:
fixed